### DISCUSSION TITLE
libtorrent import issues [Fix]

### DISCUSSION THREAD

**Discussion by Diederikjh** [Flexget/Flexget#2890] [Q&A]
I couldn't get this working for a while, and got to a solution, so want to mention it here to hopefully help someone down the line.

The symptom was that running my tasks gave the following error:

`CRITICAL task          showrss feed    libtorrent package required`

Running python in the virtualenv also failed the following statement `import libtorrent`.

Running ubuntu I thought I missed an installed package, but running python (and python3.6 and python3.8) the lib didn't work outside the virtualenv either.  (Ubuntu package `python3-libtorrent` already installed.)

When I looked closer, I saw running just normal python3 does have the correct lib (command `import libtorrent` works).

So to fix it, I had to copy the lib from the default installation dir into the virtualenv :grimacing:   Seems to work, but feels like DLL hell.

copy command:

`cp /usr/lib/python3/dist-packages/libtorrent.cpython-38-x86_64-linux-gnu.so ~/flexget/lib/python3.8/site-packages/`

Is there a better way of doing this?

**Comment by jimman2003** [comment]
Try running your virtualenv with option to enable system packages:
python -m venv (your venv dir) --site-system-packages

**Reply by Diederikjh** [reply]
`python -m venv ~/flexget/ --site-system-packages` gives me
No module named venv

**Reply by jimman2003** [reply]
> python -m venv ~/flexget/ --site-system-packages gives me
> No module named venv
you need the virtualenv package from ubuntu repos? I run arch,so cant help..

**Comment by Diederikjh** [comment]
Thanks @jimman2003 for pointing me in the right direciton.  When creating the virtual env, I can add my system packages to it, and aslong as I use the correct python version when creating it, it works.

When I run `python3 -m venv --system-site-packages ~/flexget/` and then follow the rest of the [instructions of installation](https://flexget.com/InstallWizard/Linux), it works, and has correct libtorrent dependency.
